src/runtime/components/forms/InputMenu.vue

```diff
@@ -305,12 +305,12 @@ export default defineComponent({
 
     const label = computed(() => {
       if (!props.modelValue) {
-        return
+        return null
       }
 
       if (props.valueAttribute) {
-        const option = options.value.find(option => option[props.valueAttribute] === props.modelValue)
-        return option ? accessor(option, props.optionAttribute) : null
+        const option = props.options.find(option => (typeof option === 'object' && option !== null ? accessor(option, props.valueAttribute) : option) === toRaw(props.modelValue))
```

Comment:
Why use `props.options` instead of `options.value` here? You will loose the selection if the selected item belongs to a search 😬

Reply:
What do you mean? I don't think it makes a difference here, but please have a look at my comment here: https://github.com/nuxt/ui/pull/2228#discussion_r1768839295

Reply:
Oh, sorry, you're right.
That means that we cannot initialize the options with `props.options` without accounting for the `debouncedSearch` function either.

Is the user not meant to pass the `options` prop when using the asynch search?